pkg/controller/controllercmd/builder.go

```diff
@@ -78,6 +82,7 @@ func NewController(componentName string, startFunc StartFunc) *ControllerBuilder
 		startFunc:        startFunc,
 		componentName:    componentName,
 		observerInterval: defaultObserverInterval,
+		nonZeroExitFn:    klog.Fatal,
```

Comment:
doesn't `klog.Fatal` panic? We don't want that.

Reply:
it exit(1) which is what we want and what we use today

Reply:
It writes the stacktrace, checked the code.

Reply:
i replaced this by klog.Warning() followed by os.Exit(1).